urdf/include/urdf/model.h

```diff
@@ -55,9 +58,13 @@ class Model : public ModelInterface
 {
 public:
   /// \brief Load Model from TiXMLElement
-  URDF_EXPORT bool initXml(TiXmlElement * xml);
+  URDF_EXPORT URDF_DEPRECATED("TinyXML API is deprecated, use the TinyXML2 version instead") bool initXml(TiXmlElement * xml);
```

Comment:
Could this break someone downstream using `-Werror`? If so maybe the target branch should be melodic.

Reply:
That's a good point.  The downside to targeting a melodic branch is that we'll probably end up supporting 3 branches: `kinetic-devel` (with only tinyxml APIs), `melodic-devel` (with tinyxml and tinyxml2 APIs and the deprecation warning), and `n-turtle-devel` (with only tinyxml2 APIs).  But that is more conservative, and ensures we don't break kinetic, so I think I'll go ahead and do that.

Thanks for the review!